tools/python/mllib/__init__.py

```diff
@@ -43,43 +45,47 @@
 from . import transforms
 from . import parsers
 
+
 def transform(node, *args):
-  result = node
-  for t in args:
-    if isinstance(t, CLASS_TYPES):
-      t = t()
-    result = result.dispatch(t)
-  return result
+    result = node
+    for t in args:
+        if isinstance(t, CLASS_TYPES):
+            t = t()
+        result = result.dispatch(t)
+    return result
+
 
 class Resolver:
 
-  def __init__(self, path):
-    self.path = path
+    def __init__(self, path):
+        self.path = path
+
+    def resolveEntity(self, publicId, systemId):
+        for p in self.path:
+            fname = os.path.join(p, systemId)
+            if os.path.exists(fname):
+                source = InputSource(systemId)
+                source.setByteStream(open(fname))
+                return source
+        return InputSource(systemId)
 
-  def resolveEntity(self, publicId, systemId):
-    for p in self.path:
-      fname = os.path.join(p, systemId)
-      if os.path.exists(fname):
-        source = InputSource(systemId)
-        source.setByteStream(open(fname))
-        return source
-    return InputSource(systemId)
 
 def xml_parse(filename, path=()):
-  if sys.version_info[0:2] == (2,3):
-    # XXX: this is for older versions of python
-    source = "file://%s" % os.path.abspath(filename)
-  else:
-    source = filename
-  h = parsers.XMLParser()
-  p = xml.sax.make_parser()
-  p.setContentHandler(h)
-  p.setErrorHandler(ErrorHandler())
-  p.setEntityResolver(Resolver(path))
-  p.parse(source)
-  return h.parser.tree
+    if sys.version_info[0:2] == (2, 3):
+        # XXX: this is for older versions of python
+        source = "file://%s" % os.path.abspath(filename)
```

Comment:
@astitcher ancient remnant of Pythons long gone

Reply:
ooh, more code to remove...

Reply:
#288